src/Viewer/PageViewer.jsx

```diff
@@ -208,7 +211,15 @@ export function PageViewer({
     if (coreWorker.current) {
       coreWorker.current.onmessage = function (e) {
         // console.log('message from core', e.data)
-        if (e.data.messageType === "updateRenderers") {
+        if (e.data.messageType === "sendToast") {
```

Comment:
Since alerts can come from Core, PageViewer, and ActivityViewer, we should use the same function. Line 285 will call `sendAlert` once core sends the right message type. `sendAlert` is also called from other places in PageViewer, as well as ActivityViewer (though the latter get a different function). We should send the `sendAlert` function from ActivityViewer be sent to `PageViewer` so unify them.

The actual alert should be displayed in ActivityViewer or piped outside to the containing app. Not sure which is best. I think we should start with ActivityViewer and abstract it further if needed.